lib/email_tracking_retriever.py

```diff
@@ -135,20 +135,24 @@ def get_all_mail_folder(self) -> imaplib.IMAP4_SSL:
   def get_email_ids(self) -> Any:
     date_to_search = self.get_date_to_search()
     mail = self.get_all_mail_folder()
-    from_email_address = self.get_from_email_address()
-    status, response = mail.uid('SEARCH', None,
-                                'FROM "%s"' % from_email_address, '(UNSEEN)',
-                                '(SUBJECT "shipped")',
-                                '(SINCE "%s")' % date_to_search)
-    email_ids = response[0].decode('utf-8')
+    subject_searches = self.get_subject_searches()
 
-    return email_ids.split()
+    result = set()
+    for search_terms in subject_searches:
+      search_terms = ['(SUBJECT "%s")' % phrase for phrase in search_terms]
```

Comment:
FYI, percent string formatting is the old pre-Python 3 option.  Python 3 offers several newer options: https://realpython.com/python-string-formatting/

#3 in particular seems pretty good.

Reply:
Oh sweet, filed #92 to track.